clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy

```diff
@@ -171,6 +171,12 @@ class AutoScalingWorker {
     String launchConfigName = null
 
     if (shouldSetLaunchTemplate(asgConfig)) {
+      if(asgConfig.getAssociateIPv6Address() == null) {
+        def asgConfigEnv = asgConfig.getCredentials().getEnvironment()
+        def autoEnableIPv6 = dynamicConfigService.getConfig(Boolean.class, "aws.features.launch-templates.ipv6.${asgConfigEnv}", false)
+        asgConfig.setAssociateIPv6Address(autoEnableIPv6)
+      }
```

Comment:
@ajordens I refactored a bit so this wouldn't override any inputs from the UI/API request. This value will now be `null` if the user hasn't explicitly set it. Now this auto-assignment will only occur if launch templates are enabled, the user has not opted into IPv6, **_and_** IPv6 is auto enabled. 

Let me know if that makes sense. Basically, instead of defaulting in the `BasicAmazonDeployDescription`, I am intercepting with the config and setting here.

Reply:
This looks good to me.